### PR TITLE
Support Gradle publishing

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/407)

### CODE REVIEW

**Comment by sschuberth** [comment]
> Gradle: Add missing getRepositories() to DependencyTreeModel interface

Missing compared to what? Why did it work before?

---

Reviewed 1 of 1 files at r1.
Review status: 0 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407#-:-L7Pl1wZjjP32mWn9Y2_:b8s9vt4)*

**Comment by sschuberth** [comment]
> Gradle: Remove unnecessary empty lines from interaces

Typo: "interfaces".

---

Reviewed 1 of 1 files at r2.
Review status: 0 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407#-:-L7PlIsftlux_UILRvTZ:bkgl2bm)*

**Comment by sschuberth** [comment]
Reviewed 3 of 3 files at r3.
Review status: 3 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407)*

**Comment by sschuberth** [comment]
Reviewed 8 of 8 files at r4.
Review status: 8 of 11 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407)*

**Comment by sschuberth** [comment]
Reviewed 3 of 3 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407)*

**Comment by mnonnenmacher** [comment]
Done.

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/heremaps/oss-review-toolkit/407#-:-L7PuqNR87iVfQLX7V5F:b-896fix)*

File: analyzer/src/main/kotlin/managers/Gradle.kt [hunk 1]
```diff
@@ -193,7 +193,7 @@ class Gradle : PackageManager() {
                     rawPackage
                 }
 
-                pkg.copy(vcsProcessed = processPackageVcs(pkg.vcs))
+                pkg.copy(vcsProcessed = if (dependency.localPath != null) processProjectVcs(File(dependency.localPath), pkg.vcs) else processPackageVcs(pkg.vcs))
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Gradle.kt]: Maybe a bit nicer:

    dependency.localPath?.let { processProjectVcs(File(it), pkg.vcs) } ?: processPackageVcs(pkg.vcs)

**Comment by mnonnenmacher** [comment]
Review status: 0 of 12 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[analyzer/src/main/kotlin/managers/Gradle.kt, line 196 at r11](https://beta.reviewable.io/reviews/heremaps/oss-review-toolkit/407#-L7PEIzE79bW_cbIWHfj-r11-196:-L7Pycn8ATMSRoKpvnQs:b-896fix) ([raw file](https://github.com/heremaps/oss-review-toolkit/blob/f9925bd52b15418c51903333ac40bf6b060193d3/analyzer/src/main/kotlin/managers/Gradle.kt#L196)):*
<details><summary><i>Previously, sschuberth (Sebastian Schuberth) wrote…</i></summary><blockquote>

Maybe a bit nicer:

    dependency.localPath?.let { processProjectVcs(File(it), pkg.vcs) } ?: processPackageVcs(pkg.vcs)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/heremaps/oss-review-toolkit/407)*

**Comment by sschuberth** [comment]
Reviewed 11 of 11 files at r6, 1 of 1 files at r7, 3 of 3 files at r8, 8 of 8 files at r9, 3 of 3 files at r10, 5 of 6 files at r11, 1 of 1 files at r12.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/heremaps/oss-review-toolkit/407)*

**Pull request merged**